Documentation/test-targets-usage.md

```diff
@@ -0,0 +1,108 @@
+# Test Targets
+
+BuildTools provides test targets and related logic for publishing and running tests.  The primary entry points are the [`Test` target](https://github.com/dotnet/buildtools/blob/9d362222215063c5ffb45dffccca22797d9ab74e/src/Microsoft.DotNet.Build.Tasks/PackageFiles/tests.targets#L336) in Microsoft.DotNet.Build.Tasks and the [`CloudBuild` target](https://github.com/dotnet/buildtools/blob/9d362222215063c5ffb45dffccca22797d9ab74e/src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/CloudTest.targets#L80) in Microsoft.DotNet.Build.CloudTestTasks.  The `Test` target is used to run or archive tests locally, and the `CloudBuild` target is used to run tests in Helix.
+
+## Usage scenarios
+
+*Note: Dotnet repos typically provide `BuildAndTest` and `RebuildAndTest` targets which can be used to build and run tests from a single command, so you may wish to substitute either of those targets for `Test` in the examples below.*
+ 
+#### Run tests for a project with the default options
+
+The following command runs tests for System.Collections.Immutable.Tests.csproj using the default options.
+```
+msbuild /t:Test System.Collections.Immutable.Tests.csproj
+```
+
+#### Run a single XUnit method
+
+You can use the `XUnitOptions` property to override the options used for the XUnit command that runs the tests.  For example, the following command runs only the `System.Security.Cryptography.Pkcs.Tests.CmsRecipientCollectionTests.Twoary` method.
+
+```
+msbuild /t:Test "/p:XunitOptions=-method System.Security.Cryptography.Pkcs.Tests.CmsRecipientCollectionTests.Twoary" System.Security.Cryptography.Pkcs.Tests.csproj
+```
+
+#### Debug tests in Visual Studio
+
+1.  Open the project's solution in Visual Studio.
+2.  Set the test project as the Startup Project.
+3.  Build (or rebuild) the solution.
+4.  Press F5 to debug the project.
+
+You can modify the XUnit command which will be executed when debugging by editing the *Command line arguments* in the *Debug* section of the test project's properties, for example to debug just a single test method.  Note that these changes will be overwritten when rebuilding the project.
+
+#### Run tests for a specified TFM and/or RID
+
+You can specify the target framework moniker (TFM) on which to run tests by using the `TestTFM` property.  Similarly, the runtime ID (RID) can be specified with the `TestNugetRuntimeId` property.
```

Comment:
We won't have any notion of a TestTFM the BuildConfiguration you are building for will essentially be the TestTFM. For the TestNugetRuntimeId we be replaced with BuildConfiguration and RuntimeOS properties.